### PR TITLE
Deploy docs action

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality: auto deploy docs on release

**How to test**:
1. how to test it: release a release and check that the docs are updated (or test by moving it to trigger on push once if it looks good to the reviewer?)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: .github/workflows/build_and_publish.yml [hunk 1]
```diff
@@ -55,3 +55,17 @@ jobs:
...
 
+ deploy-docs:
+   name: Deploy Docs to GitHubIO
+   runs-on: ubuntu-latest
+   steps:
+    - name: Checkout main
+      uses: actions/checkout@v2
+
+    - name: Deploy docs
+      uses: mhausenblas/mkdocs-deploy-gh-pages@master
```
Comment (northwestwitch) [inline .github/workflows/build_and_publish.yml:66]: The sh file runned by this action looks quite complicated. I wonder if it wouldn't be easier to just run the usual commands as "run" in here instead. But fine if it works!

**Review by northwestwitch — APPROVED** [review]
Hard to test unless we deploy. I think it looks harmless enough to merge. Good to have an action to update the docs! 👍

**Comment by dnil** [comment]
> Hard to test unless we release. I think it looks harmless enough to merge. Good to have an action to update the docs! 👍 

Yes, perhaps not as bad as some other similar ones for pdf as well. Do you have a good trick for uploading though - the regular gh-deploy doesn't take a GitHub.secret does it?

**Comment by northwestwitch** [comment]
> Yes, perhaps not as bad as some other similar ones for pdf as well. Do you have a good trick for uploading though - the regular gh-deploy doesn't take a GitHub.secret does it?

if you have a secret already set I think you're good, check here: https://squidfunk.github.io/mkdocs-material/publishing-your-site/

**Comment by dnil** [comment]
> > Yes, perhaps not as bad as some other similar ones for pdf as well. Do you have a good trick for uploading though - the regular gh-deploy doesn't take a GitHub.secret does it?
> 
> if you have a secret already set I think you're good, check here: https://squidfunk.github.io/mkdocs-material/publishing-your-site/

Lets try - that would be super nice! I still have a nagging feeling you would have to do what the mhausenblas script does to pass the token indirectly to MkDocs via setting it on the remote repo. But lets see, perhaps there is some good behind the scenes automagic that will set it for us!

**Pull request merged**